sources/core.ts

```diff
@@ -458,9 +458,9 @@ export const reducers = {
 // ------------------------------------------------------------------------
 
 export type ArityDefinition = {
-    leading: number;
-    extra: number;
-    trailing: number;
+    leading: string[];
+    extra: string[] | undefined;
```

Comment:
So `undefined` is the new `Infinity` in this context, right? Can we use `null` instead of `undefined` (or maybe even better - a named object exported by the core)?

```ts
export const NoLimits = {__noLimits: true};
export const NoLimitsT = {__noLimits: true};

export type ArityDefinition = {
    extra: string[] | NoLimitsT;
};

// Then later

this.arity.extra = NoLimits;
```

Reply:
This would make more sense, yes.

Reply:
Used a `Symbol` instead. I think that makes the intent clearer.